tools/bin/hawqregister

```diff
@@ -1145,7 +1153,7 @@ def main(options, args):
 
     failure_handler = FailureHandler(conn)
     # register
-    if options.yml_config and ispartition(options.yml_config):
+    if check_file_exist(options.yml_config) and ispartition(options.yml_config):
```

Comment:
Yeah, I have to say that the condition might be confused in the if statement. However, this small confusion inherits from the original logic. The if statement says :  if there is yaml file and register partition table, do HawqRegisterPartition. Otherwise do HawqRegister. The else part contains 2 situations. 1) no yaml file 2) have yaml file and register non-partition table.  

Add the function call in the if statement doing nothing except a check. We didn't notice the might confused logic before because we all thought the yaml file exist by default. We could make it more clearly with more code. But I don't think it worthes the effort to change code if we take 1 min to understand the logic.